File: src/main/java/com/urlshortner/urlshortner/Repository/UrlRepository.java
package com.urlshortner.urlshortner.Repository;
import com.urlshortner.urlshortner.Model.UrlModel;
import org.springframework.data.cassandra.repository.AllowFiltering;
import org.springframework.data.cassandra.repository.CassandraRepository;
import org.springframework.data.cassandra.repository.Query;
import org.springframework.stereotype.Repository;
import java.util.List;

@Repository
public interface UrlRepository extends CassandraRepository<UrlModel, Long> {

    @AllowFiltering
    List<UrlModel> findByLongUrl(String longurl);

    @AllowFiltering
    List<UrlModel> findByShortUrl(String shorturl);

    @AllowFiltering
    UrlModel findUrlModelByShortUrl(String shorturl);

    @AllowFiltering
    int countUrlModelByShortUrl(String urlId);

}

File: src/main/java/com/urlshortener/urlshortener/repository/UrlRepository.kt
package com.urlshortener.urlshortener.repository

import com.urlshortener.urlshortener.model.UrlModel
import org.springframework.data.cassandra.repository.AllowFiltering
import org.springframework.data.cassandra.repository.CassandraRepository
import org.springframework.stereotype.Repository

@Repository
interface UrlRepository : CassandraRepository<UrlModel, Long> {

    @AllowFiltering
    fun findByLongUrl(longurl: String): List<UrlModel?>

    @AllowFiltering
    fun findByShortUrl(shorturl: String?): List<UrlModel?>?

    @AllowFiltering
    fun findUrlModelByShortUrl(shorturl: String?): UrlModel?

    @AllowFiltering
    fun countUrlModelByShortUrl(urlId: String?): Int
}
